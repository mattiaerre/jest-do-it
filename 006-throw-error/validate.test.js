const validate = require('./validate');

test('req is valid', () => {
  const req = {
    user: {
      name: 'john.doe@example.com'
    },
    token: '6901406e-bd3e-472b-b0f2-d4aaa250f17e'
  };

  expect(validate(req)).toBe(true);
});

test('req is not valid', () => {
  const req = {
    user: {
      name: 'john.doe@example.com'
    }
  };

  expect(() => { validate(req); }).toThrowErrorMatchingSnapshot();
});
