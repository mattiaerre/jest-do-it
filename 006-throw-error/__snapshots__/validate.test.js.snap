// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`req is not valid 1`] = `"req doesn't have a valid token"`;
